app/services/PaymentAPIService.scala

```diff
@@ -0,0 +1,78 @@
+package services
+
+import java.io.IOException
+
+import play.api.libs.json.{JsObject, JsValue, Json}
+import play.api.libs.ws.{WSClient, WSResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object PaymentAPIService {
+  case class Email(value: String)
+  object Email {
+    def fromResponse(resp: WSResponse): Either[Throwable, Option[Email]] = {
+      if (resp.status == 200) {
+        Right((resp.json \ "email").validate[String].asOpt.map(Email.apply))
+      } else {
+        Left(new IOException(resp.toString))
+      }
+    }
+  }
+}
+
+class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String, paymentApiPayPalExecutePaymentPath: String) {
+
+  def convertQueryString(queryString: Map[String, Seq[String]]): List[(String, String)] = {
+    queryString.foldLeft(List.empty[(String, String)]) {
+      case (list, (key, values)) => list ::: values.map(x => (key, x)).toList
+    }
+  }
+
+  private def postData(data: JsObject, queryStrings: Map[String, Seq[String]], isTestUser: Boolean) = {
+
+    val testQueryParam = if (isTestUser) "?mode=test" else ""
```

Comment:
Add this query param into the queryStrings map rather than using String concatenation